File: msgraph-mail/java/utilities/src/main/java/graphjavav4/utilities/users/SingleValueLegacyExtendedProperty.java
package graphjavav4.utilities.users;

import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import java.util.function.BiConsumer;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
public class SingleValueLegacyExtendedProperty extends Entity implements Parsable {
    /** A property value.  */
    @javax.annotation.Nullable
    public String value;
    /**
     * The deserialization information for the current model
     * @return a Map<String, BiConsumer<T, ParseNode>>
     */
    @javax.annotation.Nonnull
    public <T> Map<String, BiConsumer<T, ParseNode>> getFieldDeserializers() {
        return new HashMap<>(super.getFieldDeserializers()) {{
            this.put("value", (o, n) -> { ((SingleValueLegacyExtendedProperty)o).value = n.getStringValue(); });
        }};
    }
    /**
     * Serialiazes information the current object
     * @param writer Serialization writer to use to serialize this model
     * @return a void
     */
    public void serialize(@javax.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        super.serialize(writer);
        writer.writeStringValue("value", value);
    }
}

File: msgraph-mail/ruby/graphrubyv4/users/single_value_legacy_extended_property.rb
require 'microsoft_kiota_abstractions'
require_relative './entity'
require_relative './users'

module Graphrubyv4::Users
    class SingleValueLegacyExtendedProperty < Graphrubyv4::Users::Entity
        include MicrosoftKiotaAbstractions::Parsable
        ## 
        # A property value.
        @value
        ## 
        ## Gets the value property value. A property value.
        ## @return a string
        ## 
        def  value
            return @value
        end
        ## 
        ## The deserialization information for the current model
        ## @return a i_dictionary
        ## 
        def get_field_deserializers() 
            return {
                "value" => lambda {|o, n| o.value = n.get_object_value(string) },
            }
        end
        ## 
        ## Serializes information the current object
        ## @param writer Serialization writer to use to serialize this model
        ## @return a void
        ## 
        def serialize(writer) 
            super.serialize(writer)
            writer.write_object_value("value", self.value)
        end
        ## 
        ## Sets the value property value. A property value.
        ## @param value Value to set for the value property.
        ## @return a void
        ## 
        def  value=(value)
            @value = (value)
        end
    end
end
